File: src/main/scala/committee/nova/drinkit/config/CommonConfig.scala
package committee.nova.drinkit.config

import committee.nova.drinkit.config.CommonConfig.{config, drinkable}
import cpw.mods.fml.common.event.FMLPreInitializationEvent
import net.minecraftforge.common.config.Configuration

object CommonConfig {
  def init(e: FMLPreInitializationEvent): Unit = new CommonConfig(e)

  private var config: Configuration = _
  var drinkable: Array[String] = _
}

class CommonConfig(e: FMLPreInitializationEvent) {
  config = new Configuration(e.getSuggestedConfigurationFile)
  config.load()
  drinkable = config.getStringList("drinkable", Configuration.CATEGORY_GENERAL, Array("mushroomStew", "exampleDrinkable1", "exampleDrinkable2"), "List of drinkable item's unlocalized name")
  config.save()
}

File: src/main/java/committee/nova/drinkit/config/CommonConfig.java
package committee.nova.drinkit.config;

import net.minecraftforge.common.config.Configuration;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;

public class CommonConfig {
    private static Configuration cfg;
    private static String[] drinkable;

    public static void init(FMLPreInitializationEvent e) {
        cfg = new Configuration(e.getSuggestedConfigurationFile());
        cfg.load();
        drinkable = cfg.getStringList("drinkable", Configuration.CATEGORY_GENERAL, new String[]{"minecraft:mushroom_stew", "minecraft:beetroot_soup", "example_namespace:example_item"}, "List of drinkable item's resource location name");
        cfg.save();
    }

    public static String[] getDrinkable() {
        return drinkable;
    }
}
